pvlib/tests/test_singlediode.py

```diff
@@ -69,6 +71,124 @@ def test_newton_fs_495(method, cec_module_fs_495):
     assert np.isclose(pvs_ixx, ixx)
 
 
+def build_precise_iv_curve_dataframe(file_csv, file_json):
+    """
+    Reads a precise IV curve parameter set CSV and JSON to create a data frame
+    with these columns: ``Index``, ``photocurrent``, ``saturation_current``,
+    ``resistance_series``, ``resistance_shunt``, ``n``, ``cells_in_series``,
+    ``Voltages``, ``Currents``, ``v_oc``, ``i_sc``, ``v_mp``, ``i_mp``,
+    ``p_mp``, ``i_x``, ``i_xx`, ``Temperature``, ``Irradiance``,
+    ``Sweep Direction``, ``Datetime``, ``Boltzman``, ``Electron Charge``, and
+    ``Vth``. The columns ``Irradiance``, ``Sweep Direction`` are None or empty
+    strings.
+
+    Parameters
+    ----------
+    file_csv: str
+        Path to a CSV file of iv curve parameter sets.
+
+    file_json: str
+        Path to a JSON file of precise iv curves.
+
+    Returns
+    -------
+        A data frame.
+    """
+    params = pd.read_csv(file_csv)
+    curves_metadata = pd.read_json(file_json)
+    curves = pd.DataFrame(curves_metadata['IV Curves'].values.tolist())
+    curves['cells_in_series'] = curves_metadata['cells_in_series']
+    joined = params.merge(curves, on='Index', how='inner',
+                          suffixes=(None, '_drop'), validate='one_to_one')
+    joined = joined[(c for c in joined.columns if not c.endswith('_drop'))]
+
+    # parse strings to np.float64
+    is_array = ['Currents', 'Voltages', 'diode_voltage']
+    joined[is_array] = joined[is_array].applymap(
+        lambda a: np.asarray(a, dtype=np.float64)
+    )
+    is_number = joined.columns.difference(
+        is_array + ['Index', 'Irradiance', 'Sweep direction', 'Datetime']
+    )
+    joined[is_number] = joined[is_number].applymap(np.float64)
+
+    joined['Boltzman'] = 1.380649e-23
+    joined['Electron Charge'] = 1.60217663e-19
```

Comment:
For better traceability, I would think that the choice of values for these constants should documented as part of the files containing the parameters used to precisely compute the curves.

For comparison, `scipy==1.8.1` gives:

scipy.constants.value("Boltzmann constant") = 1.380649e-23 J/K
scipy.constants.value("elementary charge") = 1.602176634e-19 C

Reply:
Also see https://github.com/pvlib/pvlib-python/issues/483

Reply:
Ok, I updated these constants to match The SI Brochure (https://www.bipm.org/en/publications/si-brochure) and cited that document in the docstring. In that docstring, I also added the precision cutoff of mpmath and more detail about the precision of the IV curve data. I avoided mentioning [cwhanse/ivcurves](https://github.com/cwhanse/ivcurves), but maybe we should reference it. There is much more documentation and code that would help someone understand how the IV curve data was generated.

Also, if we use `scipy.constants`, pvlib may need to update its minimum SciPy version requirement because Boltzmann's constant changes depending on the SciPy version. The value for SciPy 1.9.3 is `1.602176634e-19 C`, which matches the value hard-coded here; however, the value is `1.6021766208e-19 C` for SciPy 1.2.0. It changes in SciPy 1.4.0, and pvlib's minimum version of SciPy is 1.2.0.

Reply:
@pvlib/pvlib-maintainer I think we should advance scipy version and use the 2019 values, which are defined to be _exact_ rather than with uncertainty.